request-manager/graph/grapher.go

```diff
@@ -244,26 +265,27 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 					return nil, nil, fmt.Errorf("node %s missing job args: %s", nodeSpec.Name, strings.Join(missingArgs, ", "))
 				}
 
-				// Insert component into graph
+				// Insert node into graph
 				if len(nodeSpec.Dependencies) == 0 {
 					// Case: no dependencies; insert directly after start node
-					err := seqGraph.InsertComponentBetween(component, seqGraph.Source, seqGraph.Sink)
+					err := seqGraph.InsertComponentBetween(node, seqGraph.Source, seqGraph.Sink)
```

Comment:
I know what's happening here and why, but it's part of the code that has always confused me. If a node has zero deps, that means it has zero in-edges, e.g. A in A->B->C. But the confusing part is: what if, in that example, we have already processed C and B, then how is A inserted between `seqGraph.Source, seqGraph.Sink`? Aren't there already 2 nodes (B and C) between source and sink?